### PR TITLE
containerapp refactor client

### PR DESCRIPTION
**Pull request by Greedygre**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

Refactor clients for separating GA and preview extension.
For containerapp extension, use `CURRENT_API_VERSION` to control version.
For containerapp-preview extension, use `PREVIEW_API_VERSION`. The command in containerapp-preview will overwrite the command in containerapp, and api-version value is `PREVIEW_API_VERSION`. https://github.com/Azure/azure-cli-extensions/pull/6419

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by Greedygre** [comment]
hi @zhoxing-ms 
I found after the pr https://github.com/Azure/azure-cli-extensions/pull/6419 merged, the pipeline check failed due to:

```

======================================================================
ERROR: test_ref_doc_containerapp-preview (__main__.IndexRefDocs)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/vss/_work/1/s/./scripts/ci/index_ref_doc.py", line 66, in test
    raise e
  File "/mnt/vss/_work/1/s/./scripts/ci/index_ref_doc.py", line 63, in test
    check_call(script_args)
  File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/opt/hostedtoolcache/Python/3.10.12/x64/bin/python', '/mnt/vss/_work/1/s/scripts/refdoc/generate.py', '--extension-file', '/tmp/tmpz06ma0oi/containerapp_preview-1.0.0b1-py3-none-any.whl', '--output-dir', '/tmp/tmpogmsf8tt/containerapp-preview']' returned non-zero exit status 1.


```

File: src/containerapp-preview/azext_containerapp_preview/tests/latest/test_containerapp_preview_scenario.py [hunk 1]
```diff
@@ -24,16 +24,6 @@ def setUp(self):
         cmd = ['azdev', 'extension', 'add', 'k8s-extension']
         run(cmd, check=True)
 
-    def tearDown(self):
-        cmd = ['azdev', 'extension', 'remove', 'containerapp']
```
Comment (Greedygre) [inline src/containerapp-preview/azext_containerapp_preview/tests/latest/test_containerapp_preview_scenario.py:28]: for tests case be more stable and faster

Comment (Greedygre) [reply src/containerapp-preview/azext_containerapp_preview/tests/latest/test_containerapp_preview_scenario.py:28]: @wangzelin007 

This CI test failed [Pipelines - Run 20230712.36 logs (azure.com)](https://dev.azure.com/azclitools/public/_build/results?buildId=71727&view=logs&jobId=ff866e8d-4875-52ff-9fd3-f181f728432c&j=ff866e8d-4875-52ff-9fd3-f181f728432c&t=64f04c5b-80b8-53a1-8045-e38ac626841b)

only python 3.10 tests failed. And Error as following.

[gw0] [ 50%] FAILED src/containerapp-preview/azext_containerapp_preview/tests/latest/test_containerapp_preview_scenario.py::ContainerappScenarioTest::test_containerapp_preview_e2e

src/containerapp-preview/azext_containerapp_preview/tests/latest/test_containerapp_preview_scenario.py::ContainerappScenarioTest::test_containerapp_preview_environment_type

[gw0] [100%] PASSED src/containerapp-preview/azext_containerapp_preview/tests/latest/test_containerapp_preview_scenario.py::ContainerappScenarioTest::test_containerapp_preview_environment_type

 

ERROR cli.azure.cli.core.azclierror:azlogging.py:212 unrecognized arguments: --environment-type managed

ERROR az_command_data_logger:azlogging.py:213 unrecognized arguments: --environment-type managed

Comment (Greedygre) [reply src/containerapp-preview/azext_containerapp_preview/tests/latest/test_containerapp_preview_scenario.py:28]: And with more and more test cases( with different file), this kind of `azdev extension add containerapp` in a case will be unstable

**Review by Juliehzl — APPROVED** [review]
approved with suggestions for tearDown

**Comment by Greedygre** [comment]
Hi @zhoxing-ms 
Could you help to review and merge this pr?

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**